lib/PEX.ts

```diff
@@ -41,8 +41,9 @@ export class PEX {
     presentation: IPresentation,
     limitDisclosureSignatureSuites?: string[]
   ): EvaluationResults {
+    const presentationDefinitionCopy: PresentationDefinitionV1 = JSON.parse(JSON.stringify(presentationDefinition));
```

Comment:
Is there a specific reason to you chose to do all the copying outside of the determineAndCastToInternalPresentationDefinition and have the determineAndCastToInternalPresentationDefinition modify the argument instead of doing the copying in that method, so that the original passed in def remains unchanged?

If this is what we want, I think we need to make a developer warning in the respective method changing the descriptor, so they are aware of the behaviour

Reply:
changed